### PR TITLE
Arbiter enhancements

### PR DESCRIPTION
**Pull request by eraly**
https://github.com/deeplearning4j/deeplearning4j/issues/6062

### CODE REVIEW

File: arbiter/arbiter-deeplearning4j/src/main/java/org/deeplearning4j/arbiter/layers/Deconvolution2DLayerSpace.java [hunk 1]
```diff
@@ -21,15 +21,19 @@
...
 import lombok.NoArgsConstructor;
+import org.deeplearning4j.arbiter.optimize.api.ParameterSpace;
+import org.deeplearning4j.arbiter.optimize.parameter.FixedValue;
 import org.deeplearning4j.nn.conf.layers.Deconvolution2D;
 
 @Data
 @EqualsAndHashCode(callSuper = true)
 @NoArgsConstructor(access = AccessLevel.PROTECTED) //For Jackson JSON/YAML deserialization
 public class Deconvolution2DLayerSpace extends BaseConvolutionLayerSpace<Deconvolution2D> {
+    protected ParameterSpace<int[]> dilation;
```
Comment (AlexDBlack) [inline arbiter/arbiter-deeplearning4j/src/main/java/org/deeplearning4j/arbiter/layers/Deconvolution2DLayerSpace.java]: Move this (and all dilation-related functionality - setLayerOptionsBuilder etc) to BaseConvolutionLayerSpace, it applies to all conv layers.

Though looks like we still need it again in SubsamplingLayerSpace... that doesn't extend BaseConvolutionLayerSpace for some reason. Let's not refactor it now though...

Comment (eraly) [reply arbiter/arbiter-deeplearning4j/src/main/java/org/deeplearning4j/arbiter/layers/Deconvolution2DLayerSpace.java]: ConvolutionLayerSpace also doesn't extend BaseConvolutionLayerSpace. Fix in next release.

File: arbiter/arbiter-deeplearning4j/src/main/java/org/deeplearning4j/arbiter/layers/BaseConvolutionLayerSpace.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ protected BaseConvolutionLayerSpace(Builder builder) {
         this.numParameters = LeafUtils.countUniqueParameters(collectLeaves());
     }
 
-    protected void setLayerOptionsBuilder(ConvolutionLayer.Builder builder, double[] values) {
+    protected void setLayerOptionsBuilder(Deconvolution2D.Builder builder, double[] values) {
```
Comment (AlexDBlack) [inline arbiter/arbiter-deeplearning4j/src/main/java/org/deeplearning4j/arbiter/layers/BaseConvolutionLayerSpace.java]: Change this to ```setLayerOptionsBuilder(Builder builder, double[] values)```

File: deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/ConvolutionLayer.java [hunk 3]
```diff
@@ -324,14 +331,14 @@ public ConvolutionLayer build() {
         }
     }
 
-    protected static abstract class BaseConvBuilder<T extends BaseConvBuilder<T>> extends FeedForwardLayer.Builder<T> {
+    public static abstract class BaseConvBuilder<T extends BaseConvBuilder<T>> extends FeedForwardLayer.Builder<T> {
```
Comment (eraly) [inline deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/ConvolutionLayer.java]: Made this public.

File: deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/ConvolutionLayer.java [hunk 4]
```diff
@@ -415,21 +423,38 @@ public T convolutionMode(ConvolutionMode convolutionMode) {
...
          * @param dilation Dilation for kernel
          */
-        public T dilation(int... dilation){
+        public T dilation(int... dilation) {
             this.dilation = dilation;
             return (T) this;
         }
 
-        /** Defaults to "PREFER_FASTEST", but "NO_WORKSPACE" uses less memory. */
+        public T kernelSize(int... kernelSize) {
```
Comment (eraly) [inline deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/ConvolutionLayer.java]: Needed to add these to use in arbiter.

**Pull request merged**